src/Traits/RefreshesFacadeInstance.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Glue\SpApi\Laravel\Traits;
+
+/**
+ * A Trait that clears the Facade's internal static reference to a
+ * resolved instance. Note that, depending on the implementation of the
+ * application binding, it is still possible for the same instance to be
+ * returned on subsequent calls to `getFacadeRoot`, such as when:
+ *   i) The object is registered as a singleton in a Service Provider;
+ *   ii) Testing helper methods like `[Facade]::shouldReceive` are used.
+ *     a) This is because the Mockery library retains its own static
+ *         references to its mocks.
+ *     b) This behavior should be desirable in most cases, as this trait is
+ *         not intended to disrupt Laravel's robust facade testing features.
+ */
```

Comment:
Ok so when using the SpApi facade majority of the time you want it to always new up a new instance of the facade and not use a prior instance? But there are times as you mentioned here in this comment such as when testing that it is okay for the same SpApi facade to be used multiple times?

Reply:
Right, just an informational note for developers! I wanted to make it clear that attaching this trait won't stop the same instance from being returned if it's bound explicitly to the container as a singleton (i.e. `$this->app->singleton(...)` instead of `$this->app->bind(...)`), or if they are using the `Facade` built-in testing helpers like `shouldReceive`. In the latter case, it seems to me that it's better that we let mock instance references to persist - otherwise things like calling `shouldReceive` on the Facade inheriting class wouldn't work.

Do you think the wording in the phpdoc is ok?

Reply:
Ok yeah thank you for the extra clarification there, I just wanted to make sure I was interpreting the note correctly. I think the wording in the phpdoc is good, the only sentence that was a little confusing to me was this one, "A Trait that clears the Facade's internal static reference to a resolved instance." But that could just be a me problem. I think I would need a little more information about the difference here being that the original facade without this trait was always referencing the same newed up instance, while this trait now makes it so the facade is a new newed up instance each time a static method is called. Again could just be me misunderstanding, but that's my thought on it. The rest of the note is clear to me though! 😁